apis/placement/v1alpah1/override_types.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpah1
+
+import (
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterResourceOverride defines a group of override policies about how to override the selected cluster scope resources
+// to target clusters.
+type ClusterResourceOverride struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ClusterResourceOverrideSpec.
+	// +required
+	Spec ClusterResourceOverrideSpec `json:"spec"`
+}
+
+// ClusterResourceOverrideSpec defines the desired state of the Override.
+type ClusterResourceOverrideSpec struct {
+	// ClusterResourceSelectors is an array of selectors used to select cluster scoped resources. The selectors are `ORed`.
+	// If a namespace is selected, ALL the resources under the namespace are selected automatically.
+	// LabelSelector is not supported.
+	// You can have 1-20 selectors.
+	// We only support Name selector for now.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=20
+	// +required
+	ClusterResourceSelectors []placementv1beta1.ClusterResourceSelector `json:"clusterResourceSelectors"`
+
+	// Policy defines how to override the selected resources on the target clusters.
+	// +required
+	Policy *OverridePolicy `json:"policy"`
+}
+
+// OverridePolicy defines how to override the selected resources on the target clusters.
+// More is to be added.
+type OverridePolicy struct {
+	// OverrideRules defines an array of override rules to be applied on the selected resources.
+	// The order of the rules determines the override order.
+	// When there are two rules selecting the same fields on the target cluster, the last one will win.
+	// You can have 1-100 rules.
```

Comment:
we only need 100 rules if we don't have templated override rules (i.e. append the cluster name to the field), right?